packages/components/src/ui/radix-dropdown/stories/index.tsx

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import type { ComponentMeta } from '@storybook/react';
+import { css } from '@emotion/react';
+
+/**
+ * Internal dependencies
+ */
+import { COLORS } from '../../../utils';
+import {
+	DropdownMenu,
+	DropdownMenuItem,
+	DropdownSubMenu,
+	DropdownMenuSeparator,
+	DropdownMenuCheckboxItem,
+	DropdownMenuGroup,
+	DropdownMenuLabel,
+	DropdownMenuRadioGroup,
+	DropdownMenuRadioItem,
+} from '../';
+import { useCx } from '../../../utils/hooks/use-cx';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { chevronRightSmall, menu, wordpress } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../../icon';
+
+const meta: ComponentMeta< typeof DropdownMenu > = {
+	title: 'Components/RadixDropdown',
+	component: DropdownMenu,
+	subcomponents: { DropdownMenuItem },
+	argTypes: {
+		// focusOnMount: {
+		// 	options: [ 'firstElement', true, false ],
+		// 	control: {
+		// 		type: 'radio',
+		// 	},
+		// },
+		// position: { control: { type: null } },
+		// renderContent: { control: { type: null } },
+		// renderToggle: { control: { type: null } },
+	},
+	parameters: {
+		actions: { argTypesRegex: '^on.*' },
+		controls: { expanded: true },
+		docs: { source: { state: 'open', excludeDecorators: true } },
+	},
+};
+export default meta;
+
+const iconButton = css`
+	all: unset;
+	font-family: inherit;
+	border-radius: 100%;
+	height: 35px;
+	width: 35px;
+	display: inline-flex;
+	align-items: center;
+	justify-content: center;
+	color: ${ COLORS.gray[ 900 ] };
+	background-color: white;
+	box-shadow: 0 2px 10px hsla( 0, 0%, 0%, 0.141 );
+
+	&:hover,
+	&:focus {
+		color: ${ COLORS.ui.theme };
+	}
+
+	&:focus {
+		box-shadow: 0 0 0 2px ${ COLORS.ui.theme };
+	}
+
+	svg {
+		fill: currentColor;
+	}
+`;
+
+const rightSlot = css`
+	margin-left: auto;
+	padding-left: 24px;
+`;
+
+export const DropdownMenuDemo = () => {
+	const [ bookmarksChecked, setBookmarksChecked ] = useState( true );
+	const [ urlsChecked, setUrlsChecked ] = useState( false );
+	const [ person, setPerson ] = useState( 'pedro' );
+
+	const cx = useCx();
+
+	const rightSlotClassName = cx( rightSlot );
+
+	return (
+		<DropdownMenu
+			trigger={
+				<button
+					className={ cx( iconButton ) }
+					aria-label="Customise options"
+				>
+					<Icon icon={ menu } size={ 20 } />
+				</button>
+			}
+			contentProps={ { sideOffset: 5 } }
+		>
+			<DropdownMenuGroup>
+				<DropdownMenuItem icon={ wordpress }>
+					New Tab <div className={ rightSlotClassName }>⌘+T</div>
+				</DropdownMenuItem>
+				<DropdownMenuItem>
+					New Window <div className={ rightSlotClassName }>⌘+N</div>
+				</DropdownMenuItem>
+				<DropdownMenuItem disabled>
+					New Private Window{ ' ' }
+					<div className={ rightSlotClassName }>⇧+⌘+N</div>
+				</DropdownMenuItem>
+
+				<DropdownSubMenu
+					trigger={
```

Comment:
Given the complexities/uncertainties of reusing `DropdownMenuItem` for submenu triggers, I think it might be simpler to export a designated submenu trigger component like so 👉 a0c62f2

As for whether there is a benefit to keeping the `asChild` setting consistent between both root menu and submenu triggers... I'm thinking probably not, upon further reflection. It's highly likely that the root menu trigger will be some kind of existing button component (i.e. already a `<button>` element), while a submenu trigger will most likely just be a designated DropdownMenu subcomponent that we export, or _maybe_ a custom div/span. In other words, there is a clear benefit for enabling `asChild` in the root menu trigger case, but no reason to do so in the submenu trigger case.

What do you think?